src/webchat/store/autoinject/autoinject-middleware.ts

```diff
@@ -0,0 +1,72 @@
+import { SocketClient } from "@cognigy/socket-client";
+import { Middleware } from "redux";
+import { StoreState } from "../store";
+import { autoInjectHandled, TAutoInjectAction, triggerAutoInject } from './autoinject-reducer';
+import { getAvatarForMessage, sendMessage } from '../messages/message-middleware';
+import { addMessage } from "../messages/message-reducer";
+import { Webchat } from "../../components/Webchat";
+
+export const createAutoInjectMiddleware = (webchat: Webchat): Middleware<unknown, StoreState> => api => next => (action: TAutoInjectAction) => {
+    switch (action.type) {
+        case 'SET_CONFIG':
+        case 'SET_CONNECTED':
+        case 'SET_OPEN': {
+            next(action);
```

Comment:
Don't you want to save the result and return it afterwards on :20, :23, :27 (also on :71)? So that a future code may rely on a convention of "dispatch() can return a value"?

Reply:
The idea here is that we are registering a "side effect" to these three events. The checks we want to do rely on the events to be fully processed, therefore we immediately call `next(action)` and use `return` later-on in order to avoid the action from being passed on twice.

After calling `next(action)`, using `api.getState()` provides us with the "up-to-date" state (the action has been processed by all middlewares and reducers), safe for further checks.

If the conditions are met after the action was processed, we dispatch an additional action as a "side-effect".